### DISCUSSION TITLE
Creation of an hollow rod

### DISCUSSION THREAD

**Discussion by manoflogic-alt** [GazzolaLab/PyElastica#425] [Q&A]
Hey everyone, I wanted to create an hollow rod with an inner diameter and an outer diameter. Is it possible ? If yes please let me know how. Also, I am trying to simulate the contacts between an hollow rod and a rod which is pulled from that hollow rod. Any leads regarding that would be appreciated. Thank you in advance

**Comment by armantekinalp** [comment]
> I wanted to create an hollow rod with an inner diameter and an outer diameter. Is it possible ? If yes please let me know how.

You can simulate a rod as if it is hallow technically. However, we do not have the implementation.  You need to change couple of rod variables to capture the effective moment of inertia, area and mass. The matrices you need to change that belong to rod are `bend_matrix` , `inv_mass_second_moment_of_inertia` (for moment of inertia), `shear_matrix` (for change in area), `mass`. I suggest reviewing [this paper](https://mattia-lab.com/wp-content/uploads/2018/06/Gazzola_RSOS_2018.pdf) for definitions. 

> Also, I am trying to simulate the contacts between an hollow rod and a rod which is pulled from that hollow rod. Any leads regarding that would be appreciated. 

We do not have a contact model for this application, you would need to write your own.
